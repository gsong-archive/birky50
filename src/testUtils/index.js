import * as faker from "faker";

const address = () =>
  [
    faker.address.streetAddress(),
    faker.address.city(),
    `${faker.address.stateAbbr()} ${faker.address.zipCode()}`,
  ].join(", ");

const card = () => ({
  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
  url: faker.internet.url(),
  price: faker.random.word(),
  address: address(),
  addressUrl: faker.internet.url(),
  phone: faker.phone.phoneNumber(),
  imgUrl: faker.image.imageUrl(),
});

const generate = {
  address,
  card,
  phoneNumber: faker.phone.phoneNumber,
  url: faker.internet.url,
  words: faker.random.words,
};

export { generate };
