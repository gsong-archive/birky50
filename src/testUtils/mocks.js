import * as cssFeatures from "../styles/cssFeatures";
import * as styleUtils from "../styles/utils";

export const mockGetBaseFontSize = {
  mock: () => (styleUtils.getBaseFontSize = jest.fn(() => 16)),
  restore: () => styleUtils.getBaseFontSize.mockRestore(),
};

export const mockSupportsGrid = bool => (cssFeatures.supportsGrid = bool);

export const mockSupportsObjectFit = bool =>
  (cssFeatures.supportsObjectFit = bool);
