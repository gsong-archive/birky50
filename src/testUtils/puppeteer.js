import "../env";

const DEFAULT_APP_URL = "http://localhost:3000";
const DEFAULT_JEST_TIMEOUT = 5000;
const DEFAULT_SCREENSHOTS_FOLDER = "screenshots/puppeteer";
const DEFAULT_SCROLL_CHECK_DELAY = 100;

const appUrl =
  process.env.APP_URL === undefined ? DEFAULT_APP_URL : process.env.APP_URL;

const jestTimeOut =
  parseInt(process.env.JEST_TIMEOUT, 10) || DEFAULT_JEST_TIMEOUT;

const screenshotsFolder =
  process.env.PUPPETEER_SCREENSHOTS_FOLDER === undefined
    ? DEFAULT_SCREENSHOTS_FOLDER
    : process.env.PUPPETEER_SCREENSHOTS_FOLDER;

const scrollCheckDelay =
  process.env.PUPPETEER_SCROLL_CHECK_DELAY || DEFAULT_SCROLL_CHECK_DELAY;

export const settings = { appUrl, jestTimeOut, scrollCheckDelay };

export const screenshot = async (imageName, page, options = {}) => {
  await page.screenshot({
    path: `${screenshotsFolder}/${imageName}.png`,
    ...options,
  });
};

const desktops = {
  "macbook-13": { width: 1280, height: 800 },
  "macbook-15": { width: 1440, height: 900 },
};

const mobiles = {
  "ipad-portrait": { width: 768, height: 1024 },
  "iphone-se-portrait": { width: 320, height: 568 },
  "iphone-8-portrait": { width: 375, height: 667 },
  "iphone-8-plus-portrait": { width: 540, height: 960 },
  "iphone-x-portrait": { width: 375, height: 812 },
};

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device.replace("portrait", "landscape")] = {
    width: viewport.height,
    height: viewport.width,
    isLandscape: true,
  };
});

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device] = { ...viewport, isMobile: true, hasTouch: true };
});

export const devices = Object.assign({}, desktops, mobiles);
