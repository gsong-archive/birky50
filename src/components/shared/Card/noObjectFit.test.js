import React from "react";

import * as emotion from "emotion";
import snapshotDiff from "snapshot-diff";
import { getStyles } from "jest-emotion";
import { render } from "react-testing-library";

import { generate } from "testUtils";
import { mockSupportsObjectFit } from "testUtils/mocks";

mockSupportsObjectFit(false);

const Card = require("./index").default;

describe("User sees a card", () => {
  const props = generate.card();
  const { container } = render(<Card {...props} />);

  test("text elements are present", () => {
    ["name", "price", "address", "phone"].forEach(prop =>
      expect(container.textContent).toMatch(props[prop])
    );
  });

  test("image is present", () => {
    const regex = RegExp(`background-image:url\\(${props.imgUrl}\\)`);
    expect(regex.test(getStyles(emotion))).not.toBe(null);
  });
});

test("Card is rendered", () => {
  const { container } = render(<Card {...snapshotProps} />);

  expect(container.firstChild).toMatchSnapshot();
});

test("Card w/o price is rendered", () => {
  const props = { ...snapshotProps };
  delete props.price;
  const { container: complete } = render(<Card {...snapshotProps} />);
  const { container: partial } = render(<Card {...props} />);
  const diff = snapshotDiff(complete.firstChild, partial.firstChild);

  expect(diff).toMatchSnapshot();
});

test("Card w/o phone is rendered", () => {
  const props = { ...snapshotProps };
  delete props.phone;
  const { container: complete } = render(<Card {...snapshotProps} />);
  const { container: partial } = render(<Card {...props} />);
  const diff = snapshotDiff(complete.firstChild, partial.firstChild);

  expect(diff).toMatchSnapshot();
});

test("Card w/o price and phone is rendered", () => {
  const props = { ...snapshotProps };
  delete props.phone;
  delete props.price;
  const { container: complete } = render(<Card {...snapshotProps} />);
  const { container: partial } = render(<Card {...props} />);
  const diff = snapshotDiff(complete.firstChild, partial.firstChild);

  expect(diff).toMatchSnapshot();
});

const snapshotProps = {
  name: "Little Big Burger",
  url: "https://example.com",
  price: "100.00",
  address: "123 Main Street",
  addressUrl: "https://url_to_map",
  phone: "503-543-2345",
  imgUrl: "https://url_to_img",
};
