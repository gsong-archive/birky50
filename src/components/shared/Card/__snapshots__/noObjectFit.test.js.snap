// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is rendered 1`] = `
.emotion-4 {
  font-weight: bold;
}

.emotion-15 {
  margin: 0.75rem;
}

.emotion-5 {
  margin: 0.75rem 0 0;
  color: hsl(0,0%,45%);
}

.emotion-14 > * {
  margin-top: 0.75rem;
}

.emotion-16 {
  border-radius: 0.25rem;
  box-shadow: 0 2px 6px 0 hsla(0,0%,0%,0.2);
}

.emotion-2 {
  color: hsl(44,56%,32%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-6 {
  font-family: "Segoe UI Emoji";
  margin-right: 0.25em;
}

.emotion-0 {
  border-radius: 0.25rem;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  display: block;
  height: 12rem;
  background-image: url(https://url_to_img);
  background-size: cover;
  background-position: center center;
}

<div
  class="emotion-16 emotion-17"
>
  <a
    href="https://example.com"
    rel="nofollow"
    class="emotion-0 emotion-1"
  />
  <div
    class="emotion-15"
  >
    <div
      class="emotion-4"
    >
      <a
        href="https://example.com"
        rel="nofollow"
        class="emotion-2 emotion-3"
      >
        Little Big Burger
      </a>
    </div>
    <p
      class="emotion-5"
    >
      100.00
    </p>
    <address
      class="emotion-14"
    >
      <div>
        <span
          class="emotion-9"
        >
          <span
            class="emotion-6"
            role="img"
            aria-label="round pushpin"
          >
            üìç
          </span>
          <a
            href="https://url_to_map"
            class="emotion-2 emotion-3"
          >
            123 Main Street
          </a>
        </span>
      </div>
      <div>
        <span
          class="emotion-9"
        >
          <span
            class="emotion-6"
            role="img"
            aria-label="phone receiver"
          >
            üìû
          </span>
          <a
            href="tel:503-543-2345"
            class="emotion-2 emotion-3"
          >
            503-543-2345
          </a>
        </span>
      </div>
    </address>
  </div>
</div>
`;

exports[`Card w/o phone is rendered 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -45,27 +45,8 @@
            >
              123 Main Street
            </a>
          </span>
        </div>
-       <div>
-         <span
-           class=\\"css-1nc9tcc-EmojiLabel-EmojiLabel\\"
-         >
-           <span
-             aria-label=\\"phone receiver\\"
-             class=\\"css-lm9lwg-EmojiLabel-EmojiLabel\\"
-             role=\\"img\\"
-           >
-             üìû
-           </span>
-           <a
-             class=\\"css-yhnxzn-Link ek86r7m0\\"
-             href=\\"tel:503-543-2345\\"
-           >
-             503-543-2345
-           </a>
-         </span>
-       </div>
      </address>
    </div>
  </div>"
`;

exports[`Card w/o price and phone is rendered 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -18,15 +18,10 @@
          rel=\\"nofollow\\"
        >
          Little Big Burger
        </a>
      </div>
-     <p
-       class=\\"css-omfnqx-Card-Card\\"
-     >
-       100.00
-     </p>
      <address
        class=\\"css-1t8qzrw-Card-Card\\"
      >
        <div>
          <span
@@ -45,27 +40,8 @@
            >
              123 Main Street
            </a>
          </span>
        </div>
-       <div>
-         <span
-           class=\\"css-1nc9tcc-EmojiLabel-EmojiLabel\\"
-         >
-           <span
-             aria-label=\\"phone receiver\\"
-             class=\\"css-lm9lwg-EmojiLabel-EmojiLabel\\"
-             role=\\"img\\"
-           >
-             üìû
-           </span>
-           <a
-             class=\\"css-yhnxzn-Link ek86r7m0\\"
-             href=\\"tel:503-543-2345\\"
-           >
-             503-543-2345
-           </a>
-         </span>
-       </div>
      </address>
    </div>
  </div>"
`;

exports[`Card w/o price is rendered 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -18,15 +18,10 @@
          rel=\\"nofollow\\"
        >
          Little Big Burger
        </a>
      </div>
-     <p
-       class=\\"css-omfnqx-Card-Card\\"
-     >
-       100.00
-     </p>
      <address
        class=\\"css-1t8qzrw-Card-Card\\"
      >
        <div>
          <span"
`;
